src/NovelRT/CMakeLists.txt

```diff
@@ -161,11 +161,38 @@ if(WIN32)
       Winmm
   )
 
+  set(CMAKE_INSTALL_PREFIX "C:/Program Files/NovelRT" CACHE PATH "..." FORCE)
+
   # I hate CMake I hate it I hate it I hate it
   # (work around for CMAKE REFUSING TO COPY NECESSARY FILES FOR DEBUGGING MAKING IT IMPOSSIBLE TO DEBUG)
-  add_custom_command(TARGET Engine POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory
-      ${CMAKE_BINARY_DIR}/thirdparty
-      $<TARGET_FILE_DIR:Engine>)
-  file(GLOB libs RELATIVE ${CMAKE_BINARY_DIR}/thirdparty ${CMAKE_BINARY_DIR}/thirdparty/*.dll)
+  #add_custom_command(TARGET Engine POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory
+  #    ${CMAKE_BINARY_DIR}/thirdparty
+  #    $<TARGET_FILE_DIR:Engine>)
+  #file(GLOB libs RELATIVE ${CMAKE_BINARY_DIR}/thirdparty ${CMAKE_BINARY_DIR}/thirdparty/*.dll)
+  file(GLOB libs ${CMAKE_BINARY_DIR}/thirdparty/*.dll)
   set_target_properties(Engine PROPERTIES RESOURCE "${libs}")
 endif()
+
+set_target_properties(Engine PROPERTIES PUBLIC_HEADER "${CMAKE_SOURCE_DIR}/include/NovelRT.h")
+
+install(
+  TARGETS Engine 
+  EXPORT novelrt
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
+  RESOURCE DESTINATION $<IF:$<BOOL:WIN32>,bin,lib>
+  PUBLIC_HEADER DESTINATION include
+)
+install(
+  DIRECTORY "${CMAKE_SOURCE_DIR}/include/NovelRT"
```

Comment:
This isn't necessary. Use PUBLIC_HEADER to specify the public headers.

Reply:
WhenI attempted this, it only copied the files themselves without maintaining the folder structure. Surely that would break our headers, no?

Reply:
It shouldn't do... Fabulist seems to have no problem with it

Reply:
Specifically, this is what's installed currently:
![Screenshot_20220214_192131](https://user-images.githubusercontent.com/21000186/153931692-189cc412-c81a-47ec-9c3c-582dcd634e74.png)
And the headers are specified like:
```cmake
set(public_headers
    include/fabulist/compiler/compiler.hpp
)
set_target_properties(compiler
    PROPERTIES
        OUTPUT_NAME fabulist-compiler
        PUBLIC_HEADER "${public_headers}"
)
```

Reply:
Oh, excuse me I forgot to check the install() command:

```cmake
install(
    TARGETS compiler
    EXPORT fabulist
    LIBRARY DESTINATION lib
    PUBLIC_HEADER DESTINATION include/fabulist/compiler
)
```

:upside_down_face:

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/7941
https://gitlab.kitware.com/cmake/cmake/-/issues/16739

Thanks, CMake. It's been a known issue since 2008. :unamused:

Reply:
Maybe it's time for that one insanely crazy idea I had to refactor the engine into separate modules just so that we're not fighting CMake at every possible moment.